framework/include/systems/SystemBase.h

```diff
@@ -224,12 +225,20 @@ class SystemBase : public libMesh::ParallelObject, public ConsoleStreamInterface
       const unsigned int state,
       Moose::SolutionIterationType iteration_type = Moose::SolutionIterationType::Time) const;
 
+  /**
+   * Returns the parallel type of the given solution state
+   */
+  libMesh::ParallelType
+  solutionStateParallelType(const unsigned int state,
+                            const Moose::SolutionIterationType iteration_type) const;
```

Comment:
Maybe this should just take a `StateArg`?

Reply:
I suggest it be consistent with these:

- `needSolutionState`
- `solutionState`
- `oldSolutionStateVectorName`

So I can make them all `StateArg` or leave them be.